airbyte-integrations/connectors/source-amazon-ads/unit_tests/test_report_streams.py

```diff
@@ -612,3 +609,43 @@ def test_streams_state_filter(mocker, config, state_filter, stream_class):
         assert params["stateFilter"] == ",".join(state_filter)
     else:
         assert state_filter is None
+
+
+@responses.activate
```

Comment:
Add a test for Sponsored Products report and `asins_keywords` and `asins_targets` record types. They're treated differently than other types.

Reply:
Have added tests. Please review